### ISSUE TITLE
External CSS stylesheets ignored if `link` element lacks `type` attribute

### ISSUE THREAD

**Issue by elmimmo** [dvschultz/99problems#28]
Kindle ignores external stylesheets in KF8 if the declaration of the `link` element lacks the `type` attribute, even though such is not a requirement in HTML5 (but was in XHTML1.1 and before).

i.e. This will **not work**:

```
<link rel="stylesheet" href="stylesheet.css"/>
```

but this will:

```
<link rel="stylesheet" href="stylesheet.css" type="text/css"/>
```

Both [HTML The Living Standard](http://developers.whatwg.org/semantics.html#attr-link-type) and [HTML5](http://www.w3.org/TR/html5/document-metadata.html#attr-link-type) specifications declare the attribute as “purely advisory”.

Interestingly, Kindle Previewer when in Kindle DX mode (but not in any other mode), which purportedly uses the Mobipocket variant inside the .mobi file, not the KF8 variant, does use the external stylesheet, which might imply that the bug does not apply to old Kindle devices.

**Comment by dvschultz** [comment]
Just guessing, but any chance this is due to the spec being XHTML5 instead of just plain HTML5? XHTML tends to require more verbose declarations.

http://www.sitepoint.com/web-foundations/differences-html-xhtml/

**Comment by elmimmo** [comment]
I do not think so, since what XHTML imposes over HTML is only the stricter generic XML syntax. `link` is not an element defined in any way by XML, and thus which attributes it requires and which ones are optional is defined by the HTML spec, XML imposing none.

**Comment by elmimmo** [comment]
Applies to Google Play Books too (as of today).

**Comment by WilliamOckham** [comment]
Not technically a bug. The XHTML spec most definitely does specify the required attributes on the link element and the type attribute is one of them:
http://www.w3.org/TR/2001/REC-xhtml-modularization-20010410/abstract_modules.html#s_linkmodule

**Comment by jstallent** [comment]
Actually, the attributes listed there are not required. It is just a list of the _possible_ attribute types. If you look at the DTD, you'll see that the entire list is marked as "IMPLIED", and none of them as "REQUIRED". http://www.w3.org/TR/2001/REC-xhtml-modularization-20010410/dtd_module_defs.html#a_module_Link

**Comment by WilliamOckham** [comment]
@jstallent You are correct. I hadn't looked at the DTD. The documentation for the XML Schema states that lists separated by commas indicate that the items are required. I should have read that one more closely.
